pkg/cloudprovider/aws/instance.go

```diff
@@ -321,20 +321,24 @@ func (p *InstanceProvider) getCapacityType(nodeRequest *cloudprovider.NodeReques
 	return v1alpha1.CapacityTypeOnDemand
 }
 
-// filterInstanceTypes is used to eliminate GPU instance types from the list of possible instance types when a
-// non-GPU instance type will work.  If the list of instance types consists of both GPU and non-GPU types, then only
-// the non-GPU types will be returned.  If it has only GPU types, the list will be returned unaltered.
+// filterInstanceTypes is used to eliminate less desirable instance types (like GPUs) from the list of possible instance types when
+// a set of more appropriate instance types would work. If a set of more desirable instance types is not found, then the original slice
+// of instance types are returned.
 func (p *InstanceProvider) filterInstanceTypes(instanceTypes []cloudprovider.InstanceType) []cloudprovider.InstanceType {
 	var genericInstanceTypes []cloudprovider.InstanceType
 	for _, it := range instanceTypes {
+		if aws.BoolValue(it.(*InstanceType).BareMetal) {
+			continue
```

Comment:
So if I had - 
```
  requirements:
    - key: "node.kubernetes.io/instance-type"
      operator: In
      values: ["m5.24xlarge", "m5.metal"]
```

I'm never getting a `m5.metal`, since we'll take it out of the list of instances being passed to Fleet? 


I can't think of why someone would have a combination of gpu and non-gpu instance constraints in a single provisioner, but it makes me wonder why we're doing any filtering at all.

Reply:
hmmm that's interesting. For GPUs, you could still have non-GPUs and GPUs mixed in instance-type requirements since the pod resource request for a GPU would do a filtering before this func. But you're right that baremetal would never be used if specified with other non-baremetal instance types that also had the resource requirements (which would be likely).  

I think the GPU case is the most convincing argument to do the filtering the way it is now. I also can't think of a reason for mixing baremetal specifically.

Reply:
You can possibly get it in the event that the m5.24xlarge is not available in spot.  Not sure if that logic works the same way for on-demand though.

Reply:
@tzneal ah you're right and it would work the same for OD